### ISSUE TITLE
Error when install into default directory

### ISSUE THREAD

**Issue by ChidaoDouyu** [JoeJoeTV/AstroTuxLauncher#9]
### Code:
```
$ python3 AstroTuxLauncher.py install
/usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (1.26.5) or chardet (5.1.0) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "

    ___         __           ______          
   /   |  _____/ /__________/_  __/_  ___  __
  / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
 / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
/_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
               L a u n c h e r               

Unofficial Astroneer Dedicated Server Launcher for Linux
v1.1.8

[23:32:14] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
[23:32:14] [Launcher/ERROR] Error while loading config file (TypeError): Object of type <enum 'EventType'> is not TOML serializable
[23:32:14] [Launcher/ERROR] Please check the config path parameter and/or config file
[23:32:14] [Launcher/INFO] Quitting...
```
p.s. sorry, it should be correct right now

### Config:
```
[launcher]
AutoUpdateServer = true
CheckNetwork = true
OverwritePublicIP = false
LogDebugMessages = false
AstroServerPath = "./AstroneerServer"
WinePrefixPath = "./winepfx"
LogPath = "./logs"
PlayfabAPIInterval = 2
ServerStatusInterval = 3.0
DisableEncryption = true

[launcher.notifications]
method = "ntfy"
name = "Chidao_DouyuServer"
```

### Environments:
x86_64
Debian11 
wine-8.0 (Debian 8.0~repack-4)
python3.11
Python Packages:
- alive_progress (3.1.5)
- chardet (5.1.0)
- colorlog (6.7.0)
- dataclasses_json (0.5.7)
- IPy (1.1)
- packaging (23.0)
- pansi (2020.7.3)
- pathvalidate (2.5.2)
- psutil (5.9.4)
- Requests (2.25.1)
- tomli (2.0.1)
- tomli_w (1.0.0)
- about-time==4.2.1 (from alive_progress) (4.2.1)
- grapheme==0.6.0 (from alive_progress) (0.6.0)
- six (1.16.0)

### Requirement: 
- The right version of urllib3 or chardet (or maybe it can be ignored, IDK).
- How to install the Launcher into default directory successfully.

### Thanks!

**Comment by JoeJoeTV** [comment]
Can you please put the output into a code box, because currently, it's hard to read

**Comment by ChidaoDouyu** [comment]
sorry about that, thanks to your generosity

**Comment by JoeJoeTV** [comment]
Is this the config before or after you started AstroTuxLauncher?

**Comment by ChidaoDouyu** [comment]
before

**Comment by JoeJoeTV** [comment]
Did it change after you ran it?

**Comment by ChidaoDouyu** [comment]
sorry, after
i've copied the explains and options together to the config file (i always did that)
but after ran, it changed to the easiest type as now

**Comment by ChidaoDouyu** [comment]
let me try to remake it and log the progress

**Comment by JoeJoeTV** [comment]
I can't reproduce the error you got on Debian 12 with Python 3.11.2 in a new Pytrhon Virtual Environment.
I am getting a different error with your config file, but that one is complaining about you not having settings for ntfy in the `launcher.toml` file, even though you set the notification method to `ntfy`.

**Comment by ChidaoDouyu** [comment]
can i add some parameters to the install code to set the config path?

**Comment by JoeJoeTV** [comment]
> can i add some parameters to the install code to set the config path?

You can set the config path with the `-c CONFIG_PATH` command line parameter (see `python3 AstroTuxLauncher.py --help`)

**Comment by JoeJoeTV** [comment]
Also make sure when writing the config file that you read the comments in the example on the github page.
Some things are additionally required, if you choose a notification method.

**Comment by ChidaoDouyu** [comment]
I don't need notification now, how I can close it instantly? At least after I have time to start my ntfy?

**Comment by JoeJoeTV** [comment]
> I changed the method to "none" This is the Error
> 
> ```
> $ python3 ./AstroTuxLauncher.py install
> /usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (1.26.5) or chardet (5.1.0) doesn't match a supported version!
>   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
> 
>     ___         __           ______          
>    /   |  _____/ /__________/_  __/_  ___  __
>   / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
>  / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
> /_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
>                L a u n c h e r               
> 
> Unofficial Astroneer Dedicated Server Launcher for Linux
> v1.1.8
> 
> [01:18:40] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
> [01:18:40] [Launcher/ERROR] Error while loading config file (ValueError): 'none' is not a valid NotificationMethod
> [01:18:40] [Launcher/ERROR] Please check the config path parameter and/or config file
> [01:18:40] [Launcher/INFO] Quitting...
> ```

Yeah, that is actually expected, as the "none" option is actually to use an empty string like this: `""` (The Github page is not clear on that and I will update it)

**Comment by ChidaoDouyu** [comment]
Thanks

**Comment by JoeJoeTV** [comment]
Does the install work now?

**Comment by ChidaoDouyu** [comment]
nope, the first error out again

**Comment by JoeJoeTV** [comment]
The one with "EventType"?

Please send me your full `launcher.toml` file again (You can just attach it as a file)

**Comment by ChidaoDouyu** [comment]
[launcher.txt](https://github.com/user-attachments/files/17153168/launcher.txt)

**Comment by JoeJoeTV** [comment]
Hmm, with that exact config file, I am not getting the error.
Could you try creating a python virtual environment in the folder venv with `python -m venv venv`.
This encapsulates the pip packages into that folder.
You then "enter" the venv with `source venv/bin/activate` and can "leave" it again with the command `deactivate`.
Please create it, enter it and do the `pip install -r requirements.txt` again.

**Comment by JoeJoeTV** [comment]
Also, did you change the extension to `.txt` or is that the extension the file is saved as?

**Comment by ChidaoDouyu** [comment]
toml or yml cannot be uploaded on github so i changed it directly

**Comment by ChidaoDouyu** [comment]
should I create the virtual envirenment under the folder of the launcher or anywhere?

**Comment by ChidaoDouyu** [comment]
I find two paths
```
/usr/lib/python3.11/venv
/usr/lib/python3.9/venv
```
Should I produce in 3.11/venv?

**Comment by JoeJoeTV** [comment]
No, don't search for it, the command will create a folder called `venv`. Just execute it in the same folder as `AstroTuxLauncher.py`.

**Comment by JoeJoeTV** [comment]
Just use the same python command you used for AstroTuxLauncher

**Comment by ChidaoDouyu** [comment]
I'm installing the requires, I found that the version of chardet (5.2.0) is different with outside (5.1.0)

**Comment by ChidaoDouyu** [comment]
```
>$ python3 ./AstroTuxLauncher.py install

    ___         __           ______          
   /   |  _____/ /__________/_  __/_  ___  __
  / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
 / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
/_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
               L a u n c h e r               

Unofficial Astroneer Dedicated Server Launcher for Linux
v1.1.8

[01:43:55] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
[01:43:55] [Launcher/INFO] Working directory: /opt/AstroTuxLauncher
[01:43:55] [DedicatedServer/INFO] Overwriting PublicIP field in Dedicated Server config...
[01:43:55] [Net/INFO] Getting IP from remote service
[01:43:55] [DedicatedServer/ERROR] Could not update PublicIP field: <urlopen error [Errno 111] Connection refused>
[01:43:55] [DedicatedServer/INFO] NOTICE: Encryption is disabled. All players that want to join the Dedicated Server have to disable encryption on their clients too
[01:43:55] [Launcher/INFO] Installing Astroneer Dedicated Server...
[01:43:55] [Launcher/INFO] DepotDownloader not found, downloading...
Downloading DepotDownloader |                                        | ⢿ 0% in 8s (~0s, 0.0%/s)
```

**Comment by ChidaoDouyu** [comment]
oops, another error appeared
Sorry but this was after I sent the last message
```
>$ /opt/AstroTuxLauncher# python -m venv venv
>$ /opt/AstroTuxLauncher# pwd
/opt/AstroTuxLauncher
>$ /opt/AstroTuxLauncher# cd ./venv
>$ /opt/AstroTuxLauncher/venv# ls
bin  include  lib  lib64  pyvenv.cfg
>$ /opt/AstroTuxLauncher/venv# cd ..
>$ /opt/AstroTuxLauncher# source venv/bin/activate
(venv) >$ /opt/AstroTuxLauncher# pip install -r requirements.txt
Collecting alive_progress
  Using cached alive_progress-3.1.5-py3-none-any.whl (75 kB)
Collecting chardet
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f223ab1bb90>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /packages/38/6f/f5fbc992a329ee4e0f288c1fe0e2ad9485ed064cac731ed2fe47dcc38cbf/chardet-5.2.0-py3-none-any.whl
  Downloading chardet-5.2.0-py3-none-any.whl (199 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.4/199.4 kB 21.7 kB/s eta 0:00:00
Collecting colorlog
  Downloading colorlog-6.8.2-py3-none-any.whl (11 kB)
Collecting dataclasses_json
  Downloading dataclasses_json-0.6.7-py3-none-any.whl (28 kB)
Collecting IPy
  Using cached IPy-1.1-py3-none-any.whl
Collecting packaging
  Downloading packaging-24.1-py3-none-any.whl (53 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.0/54.0 kB 11.6 kB/s eta 0:00:00
Collecting pansi
  Using cached pansi-2020.7.3-py2.py3-none-any.whl (10 kB)
Collecting pathvalidate
  Downloading pathvalidate-3.2.1-py3-none-any.whl (23 kB)
Collecting psutil
  Downloading psutil-6.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (290 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 290.5/290.5 kB 22.0 kB/s eta 0:00:00
Collecting Requests
  Downloading requests-2.32.3-py3-none-any.whl (64 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.9/64.9 kB 14.0 kB/s eta 0:00:00
Collecting tomli
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting tomli_w
  Downloading tomli_w-1.0.0-py3-none-any.whl (6.0 kB)
Collecting about-time==4.2.1
  Using cached about_time-4.2.1-py3-none-any.whl (13 kB)
Collecting grapheme==0.6.0
  Using cached grapheme-0.6.0-py3-none-any.whl
Collecting marshmallow<4.0.0,>=3.18.0
  Downloading marshmallow-3.22.0-py3-none-any.whl (49 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.3/49.3 kB 13.7 kB/s eta 0:00:00
Collecting typing-inspect<1,>=0.4.0
  Downloading typing_inspect-0.9.0-py3-none-any.whl (8.8 kB)
Collecting six
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.3.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (140 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.3/140.3 kB 20.1 kB/s eta 0:00:00
Collecting idna<4,>=2.5
  Downloading idna-3.10-py3-none-any.whl (70 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 70.4/70.4 kB 37.1 kB/s eta 0:00:00
Collecting urllib3<3,>=1.21.1
  Downloading urllib3-2.2.3-py3-none-any.whl (126 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 126.3/126.3 kB 13.8 kB/s eta 0:00:00
Collecting certifi>=2017.4.17
  Downloading certifi-2024.8.30-py3-none-any.whl (167 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 167.3/167.3 kB 20.4 kB/s eta 0:00:00
Collecting mypy-extensions>=0.3.0
  Downloading mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Collecting typing-extensions>=3.7.4
  Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Installing collected packages: IPy, grapheme, urllib3, typing-extensions, tomli_w, tomli, six, psutil, pathvalidate, packaging, mypy-extensions, idna, colorlog, charset-normalizer, chardet, certifi, about-time, typing-inspect, Requests, pansi, marshmallow, alive_progress, dataclasses_json
Successfully installed IPy-1.1 Requests-2.32.3 about-time-4.2.1 alive_progress-3.1.5 certifi-2024.8.30 chardet-5.2.0 charset-normalizer-3.3.2 colorlog-6.8.2 dataclasses_json-0.6.7 grapheme-0.6.0 idna-3.10 marshmallow-3.22.0 mypy-extensions-1.0.0 packaging-24.1 pansi-2020.7.3 pathvalidate-3.2.1 psutil-6.0.0 six-1.16.0 tomli-2.0.1 tomli_w-1.0.0 typing-extensions-4.12.2 typing-inspect-0.9.0 urllib3-2.2.3
>$ python3 ./AstroTuxLauncher.py install

    ___         __           ______          
   /   |  _____/ /__________/_  __/_  ___  __
  / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
 / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
/_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
               L a u n c h e r               

Unofficial Astroneer Dedicated Server Launcher for Linux
v1.1.8

[01:43:55] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
[01:43:55] [Launcher/INFO] Working directory: /opt/AstroTuxLauncher
[01:43:55] [DedicatedServer/INFO] Overwriting PublicIP field in Dedicated Server config...
[01:43:55] [Net/INFO] Getting IP from remote service
[01:43:55] [DedicatedServer/ERROR] Could not update PublicIP field: <urlopen error [Errno 111] Connection refused>
[01:43:55] [DedicatedServer/INFO] NOTICE: Encryption is disabled. All players that want to join the Dedicated Server have to disable encryption on their clients too
[01:43:55] [Launcher/INFO] Installing Astroneer Dedicated Server...
[01:43:55] [Launcher/INFO] DepotDownloader not found, downloading...
Downloading DepotDownloader |⚠︎                                       | 0% in 10.2s (0.00%/s) 
[01:44:05] [Launcher/CRITICAL] Error while installing server on line 591: RemoteDisconnected: Remote end closed connection without response
[01:44:05] [Launcher/ERROR] Traceback (most recent call last):
  File "/opt/AstroTuxLauncher/./AstroTuxLauncher.py", line 591, in <module>
    launcher.update_server()
  File "/opt/AstroTuxLauncher/./AstroTuxLauncher.py", line 387, in update_server
    steam.dl_depotdownloader(path.dirname(self.depotdl_path), path.basename(self.depotdl_path))
  File "/opt/AstroTuxLauncher/utils/steam.py", line 86, in dl_depotdownloader
    zip_path, _ = dl.download(bar)
                  ^^^^^^^^^^^^^^^^
  File "/opt/AstroTuxLauncher/utils/steam.py", line 64, in download
    file_path, http_msg = request.urlretrieve(self.url, filename=self.filename, reporthook=self._reporthook)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 241, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
                            ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 519, in open
    response = self._open(req, data)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/request.py", line 1352, in do_open
    r = h.getresponse()
        ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/lib/python3.11/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response
```

**Comment by JoeJoeTV** [comment]
> I'm installing the requires, I found that the version of chardet (5.2.0) is different with outside (5.1.0)

Yes, that is why you want to try inside the venv

**Comment by JoeJoeTV** [comment]
Ok, now with the latest error it seem like AstroTuxLauncher can't connect to Github to download DepotDownloader and since it also couldn't get the public IP something seems to e blocking outgoing connections.

**Comment by ChidaoDouyu** [comment]
It's so generous of you to guide me.
So how can I get unblocked?
My server is with Public v4 IP

**Comment by JoeJoeTV** [comment]
Hmm, I can't really guide you, as I don't know why the connection doesn't work
Does the following command work for you on the machine that is also running AstroTuxLauncher and download the file?
```
wget https://github.com/SteamRE/DepotDownloader/releases/latest/download/DepotDownloader-linux-x64.zip
```
(this assumes you have wget installed, if not, install it or use curl or something similar)

**Comment by ChidaoDouyu** [comment]
I can ping to github.com
```
(venv) /opt/AstroTuxLauncher# ping github.com
PING github.com (20.205.243.166) 56(84) bytes of data.
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=1 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=2 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=3 ttl=109 time=99.9 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=4 ttl=109 time=99.7 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=5 ttl=109 time=100 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=6 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=7 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=8 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=9 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=10 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=11 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=12 ttl=109 time=99.9 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=13 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=14 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=15 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=16 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=17 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=18 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=19 ttl=109 time=99.9 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=20 ttl=109 time=99.8 ms
q64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=21 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=22 ttl=109 time=99.8 ms
64 bytes from 20.205.243.166 (20.205.243.166): icmp_seq=23 ttl=109 time=99.8 ms
^C
--- github.com ping statistics ---
23 packets transmitted, 23 received, 0% packet loss, time 22022ms
rtt min/avg/max/mdev = 99.732/99.813/100.138/0.078 ms
```

**Comment by JoeJoeTV** [comment]
But does the download work?

**Comment by ChidaoDouyu** [comment]
Could it be banned because of the venv?
```
(venv) /opt/AstroTuxLauncher# wget https://github.com/SteamRE/DepotDownloader/releases/latest/download/DepotDownloader-linux-x64.zip
--2024-09-27 02:00:37--  https://github.com/SteamRE/DepotDownloader/releases/latest/download/DepotDownloader-linux-x64.zip
Resolving github.com (github.com)... 20.205.243.166
Connecting to github.com (github.com)|20.205.243.166|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.1/DepotDownloader-linux-x64.zip [following]
--2024-09-27 02:00:38--  https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.1/DepotDownloader-linux-x64.zip
Reusing existing connection to github.com:443.
HTTP request sent, awaiting response... Read error (The request is invalid.) in headers.
Retrying.

--2024-09-27 02:01:09--  (try: 2)  https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.1/DepotDownloader-linux-x64.zip
Connecting to github.com (github.com)|20.205.243.166|:443...^C
```

**Comment by JoeJoeTV** [comment]
Did the command finish successfully or not?

> Could it be banned because of the venv?

You can just deactivate the venv and try this, but it _should_ not be the case.

**Comment by ChidaoDouyu** [comment]
I deact away and test the wget code.
It seems doesn't work

The command ran for a long time so I stop it with ^C
I'll test again

**Comment by JoeJoeTV** [comment]
I mean it's downloading something, so it'll take some time.
Just let it run through

**Comment by ChidaoDouyu** [comment]
![image](https://github.com/user-attachments/assets/b94fbf99-954a-41e4-8cf9-156b490c8ff0)
This is the log outside of venv

**Comment by ChidaoDouyu** [comment]
My server is in China so maybe firewall is the reason.
I have an VPN but I don't know how to deploy it on Linux.
It's hard to make all things perfect.

It's super slow now. The speed displays as --.--KB/s.
And it hasn't finished since 2 minutes ago.

**Comment by JoeJoeTV** [comment]
Ah, maybe that could be a problem, yeah. This might also hinde you running an Astroneer Server in general, as it needs to connect to Playfab (Service by Microsoft), but I'm not sure.
But the problem sees to not just be in the venv, correct?

**Comment by JoeJoeTV** [comment]
hmm, so this is a bigger problem. I don't think I can help you too much with this, as it seems to be a general problem, not just for AstroTuxLauncher.
Did the `git clone` work though or did you download it separately?

**Comment by ChidaoDouyu** [comment]
I download the Launcher with git clone successfully.
Idk why wget is as this.

**Comment by ChidaoDouyu** [comment]
I changed the public IP config in AstroServer config file. This is the path.
`/opt/AstroTuxLauncher/AstroneerServer/Astro/Saved/Config/WindowsServer/AstroServerSettings.ini`

I deployed VPN on the machine and tested it successfully, including inside venv and outside.
```
(venv) /opt/AstroTuxLauncher# wget https://github.com/SteamRE/DepotDownloader/releases/latest/download/DepotDownloader-linux-x64.zip
--2024-09-27 13:29:37--  https://github.com/SteamRE/DepotDownloader/releases/latest/download/DepotDownloader-linux-x64.zip
Connecting to 127.0.0.1:7890... connected.
Proxy request sent, awaiting response... 302 Found
Location: https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.1/DepotDownloader-linux-x64.zip [following]
--2024-09-27 13:29:38--  https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.1/DepotDownloader-linux-x64.zip
Reusing existing connection to github.com:443.
Proxy request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/11851957/354fec9f-7565-4d70-8a67-d0944f83a14e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240927%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240927T052938Z&X-Amz-Expires=300&X-Amz-Signature=26274f847d00ec40c26c64ab227d21c67f395a1e910ab8c0f1dcf579e28e25fa&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3DDepotDownloader-linux-x64.zip&response-content-type=application%2Foctet-stream [following]
--2024-09-27 13:29:38--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/11851957/354fec9f-7565-4d70-8a67-d0944f83a14e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240927%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240927T052938Z&X-Amz-Expires=300&X-Amz-Signature=26274f847d00ec40c26c64ab227d21c67f395a1e910ab8c0f1dcf579e28e25fa&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3DDepotDownloader-linux-x64.zip&response-content-type=application%2Foctet-stream
Connecting to 127.0.0.1:7890... connected.
Proxy request sent, awaiting response... 200 OK
Length: 31587569 (30M) [application/octet-stream]
Saving to: ‘DepotDownloader-linux-x64.zip’

DepotDownloader-linux-x64.zip            100%[==================================================================================>]  30.12M  19.2MB/s    in 1.6s    

2024-09-27 13:29:41 (19.2 MB/s) - ‘DepotDownloader-linux-x64.zip’ saved [31587569/31587569]
```

And this is the result I installed the Launcher.
```
/opt/AstroTuxLauncher# source venv/bin/activate
(venv) /opt/AstroTuxLauncher# python3 AstroTuxLauncher.py install

    ___         __           ______          
   /   |  _____/ /__________/_  __/_  ___  __
  / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
 / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
/_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
               L a u n c h e r               

Unofficial Astroneer Dedicated Server Launcher for Linux
v1.1.8

[13:27:11] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
[13:27:11] [Launcher/INFO] Working directory: /opt/AstroTuxLauncher
[13:27:11] [DedicatedServer/INFO] NOTICE: Encryption is disabled. All players that want to join the Dedicated Server have to disable encryption on their clients too
[13:27:11] [Launcher/INFO] Installing Astroneer Dedicated Server...
[13:27:11] [Launcher/INFO] DepotDownloader not found, downloading...
Downloading DepotDownloader |████████████████████████████████████████| 100% in 4.6s (0.21%/s) 
[13:27:16] [Steam/INFO] Finished downloading DepotDownloader in 5.24 seconds
[13:27:16] [Launcher/INFO] Updating Astroneer Dedicated Server app from Steam...
Updating app 728470 in 32.2s 
[13:27:48] [Launcher/ERROR] Error while updating Astroneer Dedicated Server
[13:27:48] [Launcher/INFO] Goodbye!
```

Look forward to your reply.

**Comment by JoeJoeTV** [comment]
AstroTuxLauncher uses DepotDownloader to download the server from steam and it's probably not just one download URL.
Try starting the it with the `-l` option to see the output of DepotDownloader.

**Comment by ChidaoDouyu** [comment]
Yeah! A good news. Successfully installed! 
```
(venv) /opt/AstroTuxLauncher# python3 AstroTuxLauncher.py install

    ___         __           ______          
   /   |  _____/ /__________/_  __/_  ___  __
  / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
 / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
/_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
               L a u n c h e r               

Unofficial Astroneer Dedicated Server Launcher for Linux
v1.1.8

[13:45:34] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
[13:45:34] [Launcher/INFO] Working directory: /opt/AstroTuxLauncher
[13:45:34] [DedicatedServer/INFO] NOTICE: Encryption is disabled. All players that want to join the Dedicated Server have to disable encryption on their clients too
[13:45:34] [Launcher/INFO] Installing Astroneer Dedicated Server...
[13:45:34] [Launcher/INFO] Updating Astroneer Dedicated Server app from Steam...
Updating app 728470 in 4:08.6 
[13:49:43] [Steam/INFO] Finished updating app 728470 in 248.64 seconds
[13:49:43] [Launcher/INFO] Sucessfully updated Astroneer Dedicated Server to version 1.31.79.0
[13:49:43] [Launcher/INFO] Goodbye!
```

**Comment by ChidaoDouyu** [comment]
But I did nothing, only install it again. Maybe just connection waved.

**Comment by ChidaoDouyu** [comment]
So what should I do next?
Keep in the venv and start with `python3 AstroTuxLauncher.py start`?

**Comment by JoeJoeTV** [comment]
If it also works outside of the venv, you don't need to start it inside, but if it only works inside because of weird package versions, then you have to start it inside the venv.

**Comment by ChidaoDouyu** [comment]
The Server run correctly.
```
(venv) /opt/AstroTuxLauncher# python3 AstroTuxLauncher.py start

    ___         __           ______          
   /   |  _____/ /__________/_  __/_  ___  __
  / /| | / ___/ __/ ___/ __ \/ / / / / / |/_/
 / ___ |(__  ) /_/ /  / /_/ / / / /_/ />  <  
/_/  |_/____/\__/_/   \____/_/  \__,_/_/|_|  
               L a u n c h e r               

Unofficial Astroneer Dedicated Server Launcher for Linux
v1.1.8

[14:27:27] [Launcher/INFO] Configuration file path: /opt/AstroTuxLauncher/launcher.toml
[14:27:27] [Launcher/INFO] Working directory: /opt/AstroTuxLauncher
[14:27:27] [DedicatedServer/INFO] NOTICE: Encryption is disabled. All players that want to join the Dedicated Server have to disable encryption on their clients too
[14:27:29] [Launcher/INFO] No update available, the Astroneer Dedicated Server is on the newest version
[14:27:31] [Launcher/INFO] Checking Network Configuration...
[14:27:35] [Launcher/INFO] Network configuration looks good
[14:27:45] [Launcher/INFO] RCON network configuration looks good
[14:27:45] [DedicatedServer/INFO] Preparing to start the Dedicated Server...
[14:27:48] [DedicatedServer/INFO] Started Dedicated Server process (v1.31.79.0). Waiting for registration...
Waiting for Dedicated Server to register with Playfab in 35.4s 
[14:28:23] [DedicatedServer/INFO] Dedicated Server ready! Took 35.47 seconds to register
[14:28:23] [Notify/INFO] Server started!
[14:28:23] [Launcher/INFO] Enter 'help' to get help about command usage
shutdown
[14:28:46] [Notify/INFO] Server shutting down...
[14:28:49] [DedicatedServer/INFO] Dedicated Server shut down gracefully
[14:28:49] [Launcher/INFO] Goodbye!
```
BTW where should I put my local save to the server in order to make it run on the server?

**Comment by JoeJoeTV** [comment]
The save folder is in the subfolder `Astro/Saved/SaveGames` in the folder where the server files where downloaded to (by default AstroneerServer).
See the "Can I add my own saves?" section on https://astroneer.space/dedicatedserver/

**Comment by ChidaoDouyu** [comment]
Thank you about all things!
If there's no other problems I'll close this issue and delete chats unrelated to the topic.
Hope you have a nice day!

**Comment by ChidaoDouyu** [comment]
I can't join the server.
`Connecting`
`Failure to connect to server`
[AstroServerSettings.ini](http://new.puremcs.top:81/chfs/shared/AstroServerSettings.ini)
[Engine.ini](http://new.puremcs.top:81/chfs/shared/Engine.ini)

Is there any limit to the player to join the server?
My steam name is Chinese.

**Comment by JoeJoeTV** [comment]
Limit is 8 people I believe.
Setting the Max Count higher will not change anything.

For the connection problems:
Whast does AstroTuxLauncher tell you?

**Comment by ChidaoDouyu** [comment]
It put out nothing.
Silence as I didn't join in the server.

**Comment by ChidaoDouyu** [comment]
maybe let's turn to a new issue?

**Comment by JoeJoeTV** [comment]
Not as you joined the server.
Did it mention any network issues when starting?
Any warnings in the output?

**Comment by ChidaoDouyu** [comment]
There are 3 possible output about the port 7777.
```
[15:01:53] [Launcher/WARNING] The server can be reached locally, but not from outside of the local network
[15:01:53] [Launcher/WARNING] Make sure the Server Port (7777) is forwarded for UDP traffic
```
```
[15:15:33] [Launcher/WARNING] The Server is completely unreachable
[15:15:33] [Launcher/WARNING] Make sure the Server Port (7777) is forwarded for UDP traffic and check firewall settings
```
or
```
[15:06:51] [Launcher/INFO] Network configuration looks good
```

BUT my firewall config is correct.
![image](https://github.com/user-attachments/assets/a8478a01-b525-4b2d-a6b3-9c0dfe1b46fd)

**Comment by JoeJoeTV** [comment]
did you also forward the ports in your router's settings?
The server needs to be reachable poublically under the public IP and port you entered into the config files.
Also, make sure NAT loopback/hairpinning/reflection (has many names) is enabled in your router's settings, or you won't be able to connect to the server from the same PC/network.
Additionally, since this is running on Linux, you need to disable encryption on every client thatwants to connect to the server.

**Comment by ChidaoDouyu** [comment]
Umm, I'm trying to understand all you said. But...
1. the linux server is from a Cloud Server Provider and it's not a separate machine. So I can't config the router of that.
2. I also have a minecraft server on this server and it is reachable after I allow 25565 in the panel.
3. As I know that NAT is also used to economize dedicated public IP? But my server has a public IP.
4. I didn't find any settings about encryption in the client. Is that set in server-side config? I truly saw a config about client config. It is as default to enforce disable encryption. But I can't join the server. So I changed it to not enforce and test.

**Comment by ChidaoDouyu** [comment]
Also, the server is in the server center. So I always set it by Internet. Maybe I won't need to connect to it from the save network?

**Comment by ChidaoDouyu** [comment]
I abandoned.
Thank you about reply.

**Comment by JoeJoeTV** [comment]
@ChidaoDouyu Sorry that I didn't respond sooner, I was not at home and couldn't answer.


> 1. the linux server is from a Cloud Server Provider and it's not a separate machine. So I can't config the router of that.

Alright, I thought you hosted it locally on the same PC or in the same network. In that case, you just need the server's public IPv4 address (no IPv6 support sadly)

> 3. As I know that NAT is also used to economize dedicated public IP? But my server has a public IP.

I did not mean NAT itself, but NAT loopback, which just means that requests to the public IP of the network are reflected back into the network. This is only required, if your game client is in the same network as your server.

> 4. I didn't find any settings about encryption in the client. Is that set in server-side config? I truly saw a config about client config. It is as default to enforce disable encryption. But I can't join the server. So I changed it to not enforce and test.

If either the game client or the server is run using Wine on Linux, you will have to disable encryption on both the server and any client that wants to connect to the server because of something not currently working there.
This is done using a config file and AstroTuxLauncher already handles this by default for the server (you need to keep it disabled).

For the client see the following guide:

## **Step 1:** Stop the Game/Server if it is currently running

## **Step 2:** Locate the `Engine.ini` config file

- **Dedicated Server:**
    Open the root directory of the Astroneer Dedicated Server (the one that contains `AstroServer.exe`).
    The `Engine.ini` config file is located in the subfolder at the path `Astro/Saved/Config/WindowsServer`.
- **Steam Client (Windows):**
    Open the folder `"%UserProfile%/AppData/Local/Astro/Saved/Config/WindowsNoEditor"` using either Explorer or inputting it into the "Run" prompt opened by pressing ++win+r++.
    The opened folder contains the `Engine.ini` config file.
- **Steam Client (Linux using Proton):**
    Go to the path `<Your Steam Library Path>/steamapps/compatdata/361420/pfx/drive_c/users/steamuser/AppData/Local/Astro/Saved/Config/WindowsNoEditor` while replacing `Your Steam Library Path` with the path of the Steam Library that you installed Astroneer to (the default is `~/.steam/steam/steamapps`).
    The opened folder contains the `Engine.ini` config file.

## **Step 3:** Edit the `Engine.ini` config file

Open the `Engine.ini` config file in a text editor (On Windows you can simply use Notepad) and **append** the following text to the end of the file **in a new line**, ==if it is not already present==:

```ini
[SystemSettings]
net.AllowEncryption=False
```

If the text is already present in the config file, simply replace the `False`(disabled)/`True`(enabled) with the setting you want.

Then, save the file and close the text editor and folder you opened.

## **Step 4:** Start the Game/Server

Now you should have successfully disabled encryption, and you can start your game/server to test it out.

## Re-enabling Encryption

In case you want to re-enable encryption, because you want to play on a server that also has encryption enabled, simply re-open the `Engine.ini` config file and go to the lines you appended previously.
Then, replace the `False`(disabled) with `True`(enabled).
